docs/source/compiler.rst

```diff
@@ -325,25 +398,161 @@ rewiring, which is controlled by ``PRAGMA INITIAL_REWIRING``. The syntax is as f
+compiling this program with naive rewiring will **not** move the ``CZ`` to a better qubit pair:
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program('PRAGMA INITIAL_REWIRING "NAIVE"', CZ(0, 1))
+
+   print(qc.compile(p).program)
+
+   PRAGMA INITIAL_REWIRING "NAIVE"
+   CZ 0 1
+
+If, however, your program includes an instruction that does **not** use neighboring qubits the
+compiler will be required to insert swaps (virtual or real, see swaps_) that might affect the
+logical-physical qubit mapping. For example,
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program('PRAGMA INITIAL_REWIRING "NAIVE"', CZ(0, 2))
+
+   print(qc.compile(p).program)
+
+   PRAGMA INITIAL_REWIRING "NAIVE"
+   CZ 6 5
+
+In the above program ``CZ 0 2`` is not a native instruction (meaning it cannot be directly executed
+on the target device) and so the compiler must insert a swap (virtual, in this case) into the
+program. When rewiring must occur in this mode it is **not** gauranteed that the resulting program
+will have optimal fidelity.
+
+.. _partial_rewiring:
+
+PARTIAL
+^^^^^^^
+
+In this mode, the compiler begins with an empty mapping from logical qubits to physical
+qubits. During the progression of compilation this mapping will be filled-in, and thus at any point
+the mapping is said to be `partial`. Generally this gives the compiler the opportunity to assign a
+logical-to-physical qubit mapping that optimizes the fidelity of the resulting program by
+incorporating fidelity information about any qubit in the device ISA.
+
+For example, if the instruction ``CZ 0 1`` has poor fidelity, under the partial rewiring strategy
+the compiler can find an alternative that improves the program fidelity:
+
+.. code:: python
+
+   from pyquil import Program, get_qc
+   from pyquil.gates import CZ
+
+   qc = get_qc("Aspen-8", as_qvm=True)
+   p = Program('PRAGMA INITIAL_REWIRING "PARTIAL"', CZ(0, 1))
+
+   print(qc.compile(p).program)
+
+   PRAGMA INITIAL_REWIRING "PARTIAL"
+   CZ 20 27
+
+Here the compiler sees that the instruction ``CZ 20 27`` will produce a program with better fidelity
+and so opts to reassign qubits in the original program.
+
+.. _greedy_rewiring:
+
+GREEDY
+^^^^^^
+
+In this mode, the compiler chooses an initial mapping between logical and physical qubits based upon
+a greedy optimization of the `distances` between qubits used in the program and those available on
+the device. When compared to the ``PARTIAL`` strategy it is generally more efficient because it uses
+a simple heuristic; however, it will also produce a program with worse overall fidelity. If
+compilation feels too slow and you're willing to trade for fidelity, then you may see success with
+this strategy, though for most ``PARTIAL`` is recommended.
```

Comment:
would remove `though for most PARTIAL is recommended` and relocate that to the strategy heading.